tests/utils/test_process_utils.py

```diff
@@ -54,10 +46,23 @@ def test_cache_return_value_per_process():
 
     assert len({path1, path3, f2_path1, f2_path2}) == 4
 
-    queue = Queue()
-    child_proc = Process(
-        target=_test_cache_return_value_per_process_child_proc_target, args=(path1, path3, queue)
-    )
-    child_proc.start()
-    child_proc.join()
-    assert queue.get(), "Testing inside child process failed."
+    if os.name != "nt":
+        # Test child process invalidates the cache.
+        # We don't create child process by `multiprocessing.Process` because
+        # `multiprocessing.Process` creates child process by pickling the target function
+        # and start a new process to run the pickled function. But the global variable
+        # `_per_process_value_cache_map` dict content is not pickled, this make child process
+        # automatically clear the `_per_process_value_cache_map` dict content.
+        pid = os.fork()
+        if pid > 0:
+            # in parent process
+            child_pid = pid
+            # check child process exit with return value 0.
+            assert os.waitpid(child_pid, 0)[1] == 0
+        else:
+            # in forked out child process
+            child_path1 = _gen_random_str1(True)
+            child_path2 = _gen_random_str1(False)
+            test_pass = len({path1, path3, child_path1, child_path2}) == 4
```

Comment:
```suggestion
            assert len({path1, path3, child_path1, child_path2}) == 4
```

why not assert here?

Reply:
`os._exit` makes forked out child process exit immediately, in contrast, if use `assert`, no matter assert pass or raise Assertion error, the pytest testing continues running until all tests completed. But my intention is to make child process exits immediately, the remaining tests it should be run only in parent process.

Reply:
I'm running the code below with pytest:

```python
import os


def test_foo():
    os.fork()
    os._exit(0)


def test_bar():
    os.fork()
    os._exit(1)

```

Reply:
```
def test_foo():
    os.fork()
    os._exit(0)
```
your code call "os._exit" in both parent and child processes.

My code is:
```
if os.fork() > 0:
  # in parent process
  pass
else:
  # in child process
  os._exit(0)
```
```

Reply:
`os.fork` has 2 return point:
* in child process, it return 0,
* in parent process, it return the pid of child process (which is >0)

Reply:
Offline dicussed with Weichen and understood what's happening.